### ISSUE TITLE
[Question] How to go about debugging indexation failures?

### ISSUE THREAD

**Issue by oknoote** [babenkoivan/elastic-scout-driver-plus#41]
Hi,

I'm having trouble updating certain fields in the elasticsearch index for a specific model. Upon creation of the model using Eloquent a document is created in the ES index correctly containing all values passed to the Eloquent create method. 

However when then adding additional data to the model, any of the added fields+values (fields which exist as part of the ES index mapping for this model) are not indexed. The output from the toSearchableArray() method of the model appears to be as expected, containing the data that should be indexed.

I am stuck trying to find what is blocking the indexation of the added object/model information. Is anything logged at all when indexation of a document fails and if so where can I find these logs (even using the artisan scout:import command nothing seems to be logged about indexation issues)? I don't see anything appearing in the laravel.log file and there are a couple of elasticsearch log files I looked at but I can't find anything related to indexation issues which could explain documents from being updated.

@babenkoivan do you recommend any steps in investigating this problem issue further? Are indexation problems being logged at all and am I missing these somehow? Would it be an idea to log indexation errors in the laravel.log file if this is currently not the case? Not sure if anyone else is running into these kind of issues but adding a section on debugging to the documentation might be helpful.

As extra context I am working on a migration from an old SQL based search (non-Laravel) to an Elasticsearch based search using Laravel+scout+elastic-scout-driver-plus.

I hope I'm getting my point across :) Looking forward to any help you can give and let me know if you would like me to elaborate/have any questions.

Cheers. and thanks in advance,
Owen

**Comment by oknoote** [comment]
I've come a little bit further with the issue, the problem might be I'm passing empty (or possibly incorrect) values for certain fields which Elasticsearch appears to be choking on. When commenting out the to-be-updated fields in the toSearchableArray method of the model the rest is indexed correctly on saving the model. Will investigate further.

**Comment by oknoote** [comment]
I'm still trying to understand how the values I'm passing to Elasticsearch are incorrect (which seems to be the case) and looking for a way to analyse this. Any pointers in where to look are appreciated!

**Comment by chrisgrim** [comment]
Hi @oknoote 
I too was having similar issues. I think this question should probably be in the elastic-migrations git?  When you post there it would probably be helpful to see your elastic migration file, your model file and your index mapping which can be found at http://127.0.0.1:9200/modelname/_mapping

**Comment by babenkoivan** [comment]
Hi @oknoote, in general there should be an error returned by ES in case something goes wrong. Please check, that neither of your code is wrapped in try/catch block, which hides the actual error.

You can also check what parameters are passed in `$this->client->bulk($params)` in `DocumentManager::index` method. This probably will point you at the problem.

You've already mentioned, that some fields are causing the problem, what are the types of those?

**Comment by oknoote** [comment]
Thanks for helping out @babenkoivan, @chrisgrim. I seem to have found in the culprit in my case.
I was passing null values to an Elasticsearch field of type geo_point. 

The ES mapping looks as follows. From the elastic migration file:
`
$mapping->geoPoint("location_point");
`

Which results in the following mapping when querying ES:
`
"location_point": {
                    "type": "geo_point"
                },
`

This is all looking good and when passing correct geo_point values during indexation all goes well.

So for now in the toSearchableArray() function of my (Location) model I have added a check to see if the lat/long fields of the model are not null as such:
`
if($this->lat!=null && $this->long!=null)
            $merge_data["location_point"] = ["lat" => $this->lat, "lon" => $this->long];
`

Since these fields are not required this simple workaround solves the problem.

@babenkoivan I haven't wrapped any of the code in a try/catch block. The indexation issue arose while creating a new object instance of the model using Eloquent Location::create($paramsArray) function. So I'm still not sure why the laravel log didn't show any errors.

I will look at your suggestion to check what parameters are passed in $this->client->bulk($params) in DocumentManager::index.

I ran into a similar issue previously when trying to populate an ES date field, where I was passing an incorrect date format. This also took me a while to debug manually, trying to pin-point the field(s) that were causing the indexation issue.

Thanks,
Owen
